_rules/disable-shortcut-670a30.md

```diff
@@ -0,0 +1,356 @@
+
+## Expectation 2
+
+The [user interface component][] used to disable the [shortcut][keyboard shortcut] is [visible][] and [included in the accessibility tree][] with an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+This rule assumes as applicable [keyboard shortcuts][keyboard shortcut] those implemented by the test target [content][]. Any other means (e.g. browser extensions, browser settings, user agents, external browser applications) are not considered.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+- [G127 Providing a mechanism to allow users to remap or turn off character key shortcuts](https://www.w3.org/WAI/WCAG21/Techniques/general/G217)
+- [F99 Failure of Success Criterion 2.1.4 due to implementing character key shortcuts that cannot be turned off or remapped](https://www.w3.org/WAI/WCAG21/Techniques/failures/F99)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [HTML document][] has a [keyboard shortcut][] using only a [printable character][] that can be disabled by a [user interface component][] which is [visible][], [included in the accessibility tree][], and has a non-empty [accessible name][].
+
+```html
+<html>
+  <head>
+    <title>Passed Example 1 for rule 670a30</title>
+    <script src="/test-assets/ffbc54/shortcut.js"></script>
+  </head>
+  <body onload="shortcut({reference: 'singleShortcut'})">
+    <label for="text">Add to list (press + to add):</label>
+    <input type="text" id="target" />
+    <label>
+      <input type="checkbox" onclick="changeSetting('singleShortcut', 'disabled', this.checked)">
+      Turn off single character keyboard shortcut
+    </label>
+    <br />
+    <div>
+      To do list
+    </div>
+    <ul id="list"></ul>
+  </body>
+</html>
+```
+
+#### Passed Example 2
+
+The [HTML document][] has a [keyboard shortcut][] using only a [printable character][] that can be disabled by a [user interface component][] which is [visible][], [included in the accessibility tree][], and has a non-empty [accessible name][], and another [keyboard shortcut][] that requires pressing one [non-printable character][non-printable characters].
+
+```html
+<html>
+  <head>
+    <title>Passed Example 2 for rule 670a30</title>
+    <script src="/test-assets/ffbc54/shortcut.js"></script>
+  </head>
+  <body onload="shortcut({reference: 'singleShortcut'}, {shortcutKey: '«' , ctrlKey: true})">
+    <label for="text">Add to list (press + or ctrl+« to add):</label>
+    <input type="text" id="target" />
+    <label>
+      <input type="checkbox" onclick="changeSetting('singleShortcut', 'disabled', this.checked)">
+      Turn off single character keyboard shortcut
```

Comment:
Technically, it toggles rather than turn off (it can be used to turn on again).
```suggestion
      Toggle single character keyboard shortcut
```
(applies on several other examples)